### DISCUSSION TITLE
My interest is the creation of personalized workouts for threadmill and smart bike and use QZ app standalone

### DISCUSSION THREAD

**Discussion by juanrancho** [cagnulein/qdomyos-zwift#2242] [Q&A]
Hi,

First of all, I am very happy and impressed that I have finally found the app I have been looking for for a long time.
I do a lot of sports but always indoors in my own Gym and I'm a fan of interval workouts.

Mi equipment:

- threadmill NordicTrack T10.0 - https://www.fitnessdigital.com/cinta-de-correr-nordictrack-t10.0-1-mes-de-suscripcion-ifit-individual/p/10009946/

- smart bike - Zycle zBike:  https://zycle.eu/zbike/

As I mentioned in the subject of the email, my intention is not use QZ with other app like zwift, I want to use QZ standalone.

My doubts:

### **1.- Running Workouts:**
I have viewed this link (https://robertoviola.cloud/2023/11/20/exploring-the-versatility-of-qz-app-your-gateway-to-a-tailored-fitness-experience/) with info about the parameters to be used in .xml files to create new personalized workouts for the Threadmill and it´s fantastic but I am viewing in the examples (https://github.com/cagnulein/qdomyos-zwift/tree/master/train-programs-examples) that there are more parameters (like "forcespeed", etc... and I would like to know **all the parameters** and it´s function.

And other doubt: I am viewing that when I start a train program, my threadmill must be started, that is, if the threadmill is stopped, the train program can not initiate the threadmill but if you has started it (for example a minimum speed) the workout run ok.

### **2.- Bike workouts:**
I can view in certain examples the parameter "resistance" with a integer numeric value. is it possible to design workouts based in power? (that is, something similar to "power=200W")

### **3.- Workout editor:**
are there any graphic editor to create personalized workouts?
I have tested the zwift workout editor (https://www.zwiftworkout.com/) but when I try to create a running workout I can view parameters based in Power not in speed and It´s extrange. I can view parameters like "PowerLow", "PowerHigh","Power","Pace=0",etc...

**Comment by cagnulein** [comment]
> I have viewed this link (https://robertoviola.cloud/2023/11/20/exploring-the-versatility-of-qz-app-your-gateway-to-a-tailored-fitness-experience/) with info about the parameters to be used in .xml files to create new personalized workouts for the Threadmill and it´s fantastic but I am viewing in the examples (https://github.com/cagnulein/qdomyos-zwift/tree/master/train-programs-examples) that there are more parameters (like "forcespeed", etc... and I would like to know **all the parameters** and it´s function.

you can find all the parameters here https://github.com/cagnulein/qdomyos-zwift/blob/5bfa0bba16cfba11108182afd085d2b0fd066293/src/trainprogram.cpp#L1349
I guess they are self explanatory. In case let me know if you have any questions



> And other doubt: I am viewing that when I start a train program, my threadmill must be started, that is, if the threadmill is stopped, the train program can not initiate the threadmill but if you has started it (for example a minimum speed) the workout run ok.

yes exactly, the train program starts when the speed of the treadmill is greater than 0



> I can view in certain examples the parameter "resistance" with a integer numeric value. is it possible to design workouts based in power? (that is, something similar to "power=200W")

yes there is the power tag. check the link above about all the tags available :)

> are there any graphic editor to create personalized workouts?

@p3g4asus did this a long time ago http://80.211.67.253:8081/training.htm 
it could be updated in case you have any new requirements

Let me know

**Reply by p3g4asus** [reply]
Wow. I didn't even remember this!! :grin:

**Reply by p3g4asus** [reply]
This should definitely be updated if needed. Attribute number has grown a lot!!

**Comment by juanrancho** [comment]
Thank you for your quick response.

I have a couple of doubts:

Threadmill - running:

Viewing all the parameters, I can not view any parameter to define a speed ramp in threadmill, is it not possible?.
and interal? (in the zwift workout editor is possible using tags like "ramp duration", "IntervalsT").

Smart Bike - Cycling:

- What are the difference between "resistance" parameter and "Power"? in the case of "Power", How is the format? Power=200 means 200W? are ramps and intervals available to define?

Graphical interface:

- I would like that when I am training with QZ, I could see the graphical representation of my predesigned training program and over it, the heartrate, speed or power predefined. is it possible?
- 
thanks,

**Comment by cagnulein** [comment]
> Viewing all the parameters, I can not view any parameter to define a speed ramp in threadmill, is it not possible?.
> and interal? (in the zwift workout editor is possible using tags like "ramp duration", "IntervalsT").

in the zwo you can use the ramp duration and the intervalst. in the xml you can have simply to set the ramp manually. if you are interested in the ramp maybe using the zwo will solve the issue for you easily!



> * What are the difference between "resistance" parameter and "Power"? in the case of "Power", How is the format? Power=200 means 200W? are ramps and intervals available to define?

resistance set the resistance value in a fixed way. power use the erg mode of your bike is available trying to keep the same power even if you change cadence. power in the ZWO is % of your ftp as zwift standard defines. so if you have the ftp settings to 200W, if you set power to 1.0 you will have 200W. Yes ramps are also available in the ZWO. In the QZ xml instead power is express in watt (no need to put a W) and ramps work as for treadmill, so you have to create line by line a ramp



> * I would like that when I am training with QZ, I could see the graphical representation of my predesigned training program and over it, the heartrate, speed or power predefined. is it possible?

in the beta version i added a chart for power training in realtime. are you on android or ios?

I'm converting this in a discussion more than a ticket. it has much sense :)

**Reply by juanrancho** [reply]
- Related with the chart for power training in realtime, I am IOS. 
Can you share the way that I can test it?

- Related with the use of .zwo files for ramps and intervals, if we are talking of running on threadmill. After using the zwift workout editor I obtain:
```
<workout_file>
  <author>xx</author>
  <name>testJCRP</name>
  <description>xx</description>
  <sportType>run</sportType>
  <durationType>time</durationType>
  <tags/>
  <workout>
    <SteadyState Duration="300" Power="0.996" pace="0"/>
    <Ramp Duration="600" PowerLow="0.99" PowerHigh="1.366" pace="0"/>
    <IntervalsT Repeat="5" OnDuration="180" OffDuration="60" OnPower="1.592" OffPower="0.656" pace="2"/>
    <SteadyState Duration="300" Power="0.744" pace="0"/>
    <IntervalsT Repeat="3" OnDuration="35" OffDuration="120" OnPower="1.848" OffPower="0.5" pace="2"/>
    <Cooldown Duration="295" PowerLow="0.434" PowerHigh="0.25" pace="0"/>
  </workout>
</workout_file>
```
Doubts:
- It appears "Power" instead "speed" (¿?). I supose that I must enter (speed="8")?
- how can I indicate a ramp based in speed of the Threadmill?. As you can see, it appears "onpower" and "offpower", but nothing related with speed of the threadmill...
- as summary: could you share with me an example with:
1.- a simple interval of 5 mins with constant speed of 8 km/h
2.- a ramp of 5 mins with speed from 8 to 12 km/h 
3.- an interval of 3x (2mins at 10 km/h + 1 min at 8 km/h)

thanks in advance.

**Reply by cagnulein** [reply]
> * Can you share the way that I can test it?

beta sent, do a power zone workout with your bike and you will see the chart icon in the top right corner!

> It appears "Power" instead "speed" (¿?). I supose that I must enter (speed="8")?

no it's fine, it's handled. be sure to set the right pace setting under training program settings in QZ. These are used for the ZWO files (pace value)

> how can I indicate a ramp based in speed of the Threadmill?. As you can see, it appears "onpower" and "offpower", but nothing related with speed of the threadmill...

in that line is doing a ramp using the pace index 2, so give it a try it will reflect what you saw in the editor :)

> as summary: could you share with me an example with:

i guess i already answered you with the 2 answers above here

**Reply by juanrancho** [reply]
Thank You Cagnulein,

sorry for the basic question: where can I obtain the beta that was sent to me?

**Comment by p3g4asus** [comment]
When using row attribute in xml file, do not forget to surround attribute values with double quotes (eg **power="200"** and _not power=200_).

**Comment by cagnulein** [comment]
I sent you by email. Check the spam folder too, it's an email from Apple

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 19 mar 2024 alle 21:25 juanrancho ***@***.***>
ha scritto:

**Reply by juanrancho** [reply]
received, thank you

**Reply by juanrancho** [reply]
Hi,

recently I have decided to change my iphone by Samsung zFolder 5.
Due that I am now with Android and I would need the Android beta, can you send me?

**Reply by cagnulein** [reply]
hi @juanrancho you can join the beta directly from the play store page of QZ!

**Reply by juanrancho** [reply]
do you refer to QZ AI companion? (It seems that it´s not a beta due it has price)

**Reply by cagnulein** [reply]
@juanrancho no i mean this one https://play.google.com/apps/testing/org.cagnulen.qdomyoszwift

**Reply by juanrancho** [reply]
Thank you @cagnulein for your quick responses.
Sorry if I bother you but I have a lot of time waiting an app like QZ and I am very interested.

 I would like to ask you 1 more questions:

1.-  how could I do to get my smart bike connect with QZ app.
My bike is:  smart bike - Zycle zBike: https://zycle.eu/zbike/
Normally I don´t have problems connecting with zwift, Trainerday, etc...
(Related with that, I would like to say that the way Trainer day present graphically the information is very useful, I don´t know if you are planning something similar in next versions. I mean: that you could view the planned workout and over that, the performance of your actual trainning)

**Comment by cagnulein** [comment]
Install the free app nrf connect and show me a screenshot about the bike
and I will let you know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 27 mar 2024 alle 01:44 juanrancho ***@***.***>
ha scritto:

**Reply by juanrancho** [reply]
I send you screenshots:


![zBike (2)](https://github.com/cagnulein/qdomyos-zwift/assets/163849058/8d9c1c8b-4c8c-4e70-94b6-5ec10ab381ec)
![zBike (3)](https://github.com/cagnulein/qdomyos-zwift/assets/163849058/6a1dff19-ace0-40c0-bf7e-18ccf389047d)
![zBike (1)](https://github.com/cagnulein/qdomyos-zwift/assets/163849058/02c01ed2-6e00-4d96-b512-10a70b15d064)
![zBike (4)](https://github.com/cagnulein/qdomyos-zwift/assets/163849058/2e655cfe-b025-4d0a-b86b-848b48753fac)
![zBike (5)](https://github.com/cagnulein/qdomyos-zwift/assets/163849058/67ca7d0b-d323-4d74-b0e6-32309c98e5db)

**Reply by cagnulein** [reply]
perfect! select it under ftms bike setting under bike settings and you 're good to go! i'm adding as well to the compatibility list! thanks!

**Comment by juanrancho** [comment]
Hi @cagnulein, 
I have tested my zBike as you suggested me (under  ftms setting) and initially all was right:
- bike detected by QZ and connected
- I dont´use any tranning program, only in real time changing power manually from QZ tile
- after 30 secs (aprox) the bike resistance decrease because bike and QZ disconnect.
I am attaching 2 screenshots, in one of them you can view that I set 140W as T.power (I understand that this si the power that you want to use) and in "Watt" tile (I understand that it´s the power that the bike is reproducing) you can see 141W, that is, bike is reproducing the power that I introduced, but in the second screeenshot you can see that suddendly the "watt" tile decrease to 95W that means a disconection between QZ app and bike. If I change the "T.Power" quantity pressing "+" or "-" the bike connect again but in 30 secs, the disconnection occurs again.

note:
- I have selected "zbike" as power device and cadence device

can you help me?

**Reply by cagnulein** [reply]
@juanrancho Please recreate the issue and share the debug log https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

Then I can check it

Thanks

**Comment by juanrancho** [comment]
[debug-Sat_Apr_6_19_53_29_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14894505/debug-Sat_Apr_6_19_53_29_2024.log)
Here is the log file

thanks

**Reply by cagnulein** [reply]
I guess I found the issue : remove the bike from both the cadence sensor setting and also the power sensor. Press OK on both and let me know and restart qz

**Comment by cagnulein** [comment]
Thanks I will check it next week

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 6 apr 2024 alle 20:04 juanrancho ***@***.***>
ha scritto:
